libcli.c

```diff
@@ -1128,17 +1129,18 @@ int cli_loop(struct cli_def *cli, int sockfd) {
         cli->showprompt = 0;
       }
 
-      FD_ZERO(&r);
-      FD_SET(sockfd, &r);
+      memset(&pfd, 0, sizeof(pfd));
+      pfd.fd = sockfd;
+      pfd.events = POLLIN;
 
-      if ((sr = select(sockfd + 1, &r, NULL, NULL, &tm)) < 0) {
+      if ((pr = poll(&pfd, 1, tm.tv_sec)) < 0) {
```

Comment:
This line has some errors, it should be fixed. poll(2) takes timeout argument as milliseconds.
it should be tm.tv_sec * 1000 instead of plain tm.tv_sec